sqlair.go

```diff
@@ -8,6 +8,9 @@ import (
 	"github.com/canonical/sqlair/internal/expr"
 )
 
+// sqlair-provided M-type map.
+type M = map[string]any
```

Comment:
I'm not sure what this doc adds, perhaps it would be helpful to hint at its use with something like this?
```
// M represents an arbitrary number of arbitrarily-typed values. Sqlair can
// populate it with query results.
```